docs/packages/TypeScript_ESLint.mdx

```diff
@@ -50,31 +53,52 @@ export default tseslint.config(
 );
 ```
 
+This config file exports a flat config that enables both the [core eslint recommended config](https://www.npmjs.com/package/@eslint/js) and [our recommended config](../users/Shared_Configurations.mdx#recommended).
+
+For more information on the `tseslint.config` function [see below](#config).
+
+### Advanced usage
+
 You can also declare our plugin and parser in your config via this package. For example this config would enable our plugin, our parser, and type-aware linting with a few of our popular type-aware rules:
 
 ```js title="eslint.config.js"
 // @ts-check
 
+import eslint from '@eslint/js';
+import importPlugin from 'eslint-plugin-import';
 import tseslint from 'typescript-eslint';
 
-export default tseslint.config({
-  plugins: {
-    '@typescript-eslint': tseslint.plugin,
+export default tseslint.config(
+  {
+    // config with just ignores is the replacement for `.eslintignore`
+    ignores: ['**/build/**', '**/dist/**', 'src/some/file/to/ignore.ts'],
   },
```

Comment:
**question:** I applaud the idea of starting the example config with a global ignore in the new flat convention but is this syntax correct? I'm getting an `Object literal may only specify known properties, and 'ignores' does not exist in type 'ConfigWithExtends'.` error while attempting to refactor my own config.

Reply:
@pete-willard try updating to 7.0.2 first which included a fix so that TS could resolve the types correctly.

This is correct and should work just fine - it's what we do internally:
https://github.com/typescript-eslint/typescript-eslint/blob/677e7cc97c4f79cb737a4d01722777352b77b9d9/eslint.config.mjs#L45-L63

I picked it as a good example of "a separate config object" and thought it was a great example cos almost every codebase I've worked on has a `.eslintignore`!

Reply:
It is a great example to have 👍🏻 I just double checked (reinstalled packages, rebooted VSCode) and I am on the .2 version and still getting that error. Mind you my config is a `.js` file with the `@ts-check` directive. I'll try more tinkering in the morning. Sorry for hijacking the comment 😅

Reply:
I just tested this locally and it worked.
I also added a passing test case to our integration test:
https://github.com/typescript-eslint/typescript-eslint/blob/bjz-rework-docs-20240219/packages/integration-tests/fixtures/flat-config-types/eslint.config.js#L21-L24
(this test runs to typecheck and validate we haven't broken the most barebones setup).

If you can create an isolated repro repo and file an issue - I'm happy to help investigate :)